packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -90,15 +90,14 @@ final class AssetTransformer {
           await tempOutputFile.copy(outputPath);
         } else {
           tempInputFile = tempOutputFile;
-          tempOutputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath(i+2));
+          tempOutputFile = newTempFile(i+2);
           ErrorHandlingFileSystem.deleteIfExists(tempOutputFile);
```

Comment:
is this needed, given the `finally` delete at line 100?

Reply:
One of the requirements of an asset transformer is that it must produce an output file. The only (reliable and easy) way to be sure that a transformer did produce an output file is to make sure it didn't already exist before the transformer was called. This is why we have this preparatory delete call (and the one on [L70](https://github.com/flutter/flutter/blob/747d76dd1acc59c31766da6c8193dfd27d7fcb4a/packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart#L70)).

Regardless, your question identifies that this isn't as clear as it should be. I've pushed a new commit, [move preparatory deleting of output file closer to the transformation…](https://github.com/flutter/flutter/pull/151932/commits/07439e25f5c06f0a31a05d5c5af047ec65bc8056), that hopefully improves this. Let me know what you think.

Reply:
:+1: